js/main.js

```diff
@@ -14,20 +108,85 @@ var formElement = document.querySelector('form');
+      if (data.entries[i].ID.toString() === data.editing) {
+        data.entries[i].title = titleEl.value;
+        data.entries[i].url = urlEl.value;
+        data.entries[i].notes = notesEl.value;
+        // console.log(data.entries[i]);
+        // console.log('index number:', i);
+
+        // entriesListEl.children[i].find('img').setAttribute('src', urlEl.value);
+        // entriesListEl.children[i].childNodes[i].childNodes[i].setAttribute('src', urlEl.valiue);
+        entriesListEl.children[i].childNodes[0].childNodes[0].setAttribute('src', urlEl.value);
+        entriesListEl.children[i].childNodes[1].childNodes[0].textContent = titleEl.value;
+        entriesListEl.children[i].childNodes[1].childNodes[1].textContent = notesEl.value;
+      }
+    }
+
+    data.editing = null;
+
+  } else if (data.editing === null) {
+    var inputValue = {
+      title: titleEl.value,
+      url: urlEl.value,
+      notes: notesEl.value,
+      ID: data.nextEntryId
+    };
 
-  data.entries.unshift(inputValue);
-  data.nextEntryId++;
+    data.entries.unshift(inputValue);
+    data.nextEntryId++;
+
+    entriesListEl.prepend(createEntry(data.entries[0]));
+  }
+
+  // console.log(data.editing);
+
+  newEntryImgEl.setAttribute('src', 'images/placeholder-image-square.jpg');
   formElement.reset();
+
+  var entriesDivEl = document.querySelector('#entries-container');
+  entriesDivEl.setAttribute('class', 'container');
+
+  var entriesFormEl = document.querySelector('#entries-form-container');
+  entriesFormEl.setAttribute('class', 'hidden');
+
+  data.view = 'entries';
+
+  // entriesListEl.prepend(createEntry(data.entries[0]));
+  // data.view = 'entries';
+}
+
+document.addEventListener('submit', save);
+
+function clicksOnParentOfEntriesList(event) {
+  if (event.target.tagName === 'I') {
+    // console.log(event.target.parentNode.parentNode);
+    // console.log(event.target.parentNode.parentNode.getAttribute('data-entry-id'));
+    data.editing = event.target.closest('li').getAttribute('data-entry-id');
+    // var entryID = event.target.closest('li').getAttribute('data-entry-id');
+    // console.log('clicksOnParentEntriesList event fired', entryID);
+    // console.log('data.editing value:', data.editing);
+    // console.log(event.target.previousSiblingElement);
+    // console.log(event.target);
+    //  console.log(event.target.parentNode.childNodes[0].textContent);
+    // console.log(event.target.closest('li').childNodes[0].childNodes[0].currentSrc);
+    formView();
+
+    formElement['0'].value = event.target.parentNode.childNodes[0].textContent;
+    formElement['1'].value = event.target.closest('li').childNodes[0].childNodes[0].currentSrc;
+    formElement['2'].value = event.target.parentNode.childNodes[1].textContent;
+    formElement.children[1].children[0].childNodes[1].setAttribute('src', formElement['1'].value);
+    formElement.childNodes[1].childNodes[1].childNodes[1].textContent = 'Edit Entry';
```

Comment:
This form of finding elements where you drill through indexes is completely reliant on your HTML staying the same shape always. You should use `querySelector` instead so that your code is only reliant on the configuration of the element and any classes or ids applied to it.

Reply:
![image](https://user-images.githubusercontent.com/99709530/163533628-3b12f709-f77d-42a0-a46b-edf58328e9bd.png)